### ISSUE TITLE
PathSpec.match_file() returns None since 0.12.0

### ISSUE THREAD

**Issue by adrienverge** [cpburnz/python-pathspec#84]
Hello @cpburnz, and thanks for maintaining pathspec :+1:

Since release 0.12.0 and the change of signature of `PathSpec.match_file()` from `bool` to `Tuple[Optional[bool], Optional[int]]`, [yamllint](https://github.com/adrienverge/yamllint/) tests fail. New errors looks like:
```python
self.assertEqual(c.ignore.match_file('test.yaml'), False)
AssertionError: None != False
```
(↑ `c.ignore` is an instance of `pathspec.PathSpec`)

These seem easy to fix, but I wanted to ask first. I don't understand why `None` is a better return value? I guessed maybe it's a small bug?

I look at all commits in this new release and found [92a9066 "Improve debugging"](https://github.com/cpburnz/python-pathspec/commit/92a9066197f26f95a0438a70b2c8f0373c37125c), I read it but it's not clear to me.

Looking forward to know more about this. Thanks! :slightly_smiling_face:

**Comment by cpburnz** [comment]
That's a bug. Only the signature of PathSpec._match_file() (with leading underscore) was changed. .match_file() was not supposed to change. I suspect I'm returning None from .match_file() when it used to return False due to forwarding the logic to another function.

-------- Original Message --------

**Comment by cpburnz** [comment]
This is now fixed in release v0.12.1. `PathSpec.match_file()` was incorrectly returning `None` when no patterns matched the file (rather than being excluded/rejected). It now properly returns `False` as before.
